### PR TITLE
Part4

### PR DESCRIPTION
**Pull request by austin-lolli**
Here is my Project 4 Part 4 assignment, ready for review!

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -127,17 +155,152 @@ struct IntType
...
         int* heapInt = nullptr;
+        IntType& powInternal( int x );
 };
 
+//point constructor implementations
+Point::Point(FloatType& a, FloatType& b) : x(a), y(b) {}
+
+Point::Point(DoubleType& a, DoubleType& b) : x( static_cast<float>(a) ), y( static_cast<float>(b) ) {}
+
+Point::Point(IntType& a, IntType& b) : x( static_cast<float>(a) ), y( static_cast<float>(b) ) {}
```
Comment (matkatmusic) [inline main.cpp]: All of these should call `Point(float, float)`.  You can look up Forwarding Constructors if you want to learn more: 
```
Point::Point(FloatType& a, FloatType& b) : Point(static_cast<float>(a), static_cast<float>(b) ) { } 
``` 
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -127,17 +155,152 @@ struct IntType
...
+    y *= static_cast<float>(m);
+    return *this;
+}
+
+Point& Point::multiply( IntType& m )
+{
+    x *= static_cast<float>(m);
+    y *= static_cast<float>(m);
+    return *this;
+}
```
Comment (matkatmusic) [inline main.cpp]: all of these should call `multiply(float)` internally.  Look at what `Point::multiply(float)` returns.
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
A few things need changing.

File: main.cpp (hunk 2 above)
Comment (austin-lolli) [reply main.cpp]: Fixed, but a question on line 186... without the static cast on the float type reference I get an infinite function warning, even though the float operator is implemented in the FloatType class. 
Is the static cast required so the compiler knows which version of the overloaded multiply function to call (the base one that accepts a float vs the UDT Float Type)?

File: main.cpp [hunk 3]
```diff
@@ -173,32 +173,27 @@ struct IntType
...
-    return *this;
+    return Point::multiply(static_cast<float>(m));
 }
 
 Point& Point::multiply( IntType& m )
 {
-    x *= static_cast<float>(m);
-    y *= static_cast<float>(m);
-    return *this;
+    return Point::multiply(static_cast<float>(m));
```
Comment (matkatmusic) [inline main.cpp]: you don't need to fully qualify the name.  it's not a base class function.  it's another member function
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**